### DISCUSSION TITLE
Quadratic Reporting Mechanics for Addressing Mass-Produced Template Flips and Halting Bad Actor Mining

### DISCUSSION THREAD

**Discussion by ltraveler** [idena-network/idena-docs#173] [IIPs]
The correct report should halt the mining process of the bad actor identity for the entire epoch. Identities that have reported the bad flip should receive all validation rewards from the reported identity, distributed in quadratic proportion based on the stake of each identity who took part in a report committee. As of now, the validation rewards of the bad actor simply merges with the overall validation rewards pool. I believe this kind of behavior would motivate users to monitor farmers' template flips more tightly.

**Comment by teavar** [comment]
imho something like I proposed in tg https://t.me/IdenaNetworkPublic/355310

> something like N-number of reported flips across N-number of validations = suspended

**Reply by teavar** [reply]
also your comment here: https://github.com/idena-network/idena-docs/discussions/131#discussioncomment-4925803

**Comment by do2050** [comment]
Great, you really want everyone to withdraw from IDENA.
  All those complications and you want to put more complications.
 People without anything make a report for no reason just because it rewards, and most people don't earn anything for just one flip, all that and you want to put more penalties. Let me tell you good news, everyone will leave the project, Congratulations

**Reply by ltraveler** [reply]
Well, if we try to set aside populism and instead focus on identifying the source of the problem, we can see that there are two significant issues that could potentially harm the blockchain. One of these issues is AI, which is another subject entirely. The second one involves farmers. If we view the blockchain as a complete system, we can observe that each member must be motivated to filter out mass-produced flips created from templates.
If you've already read my article [The Hollow Ring of iDNA Burn in Proposed IIP-13 Changes](https://medium.com/@idna.project/19d8c3e5662d), where I explain why validation rewards of reported identities should not go to the 0 wallet fund, we can use that rewards to encourage people to perform their filtering tasks more effectively. If this IIP were to be accepted, individuals would have a personal interest in doing a better job of filtering considering the fact that you would earn rewards proportionate to your stake.
The potential loss of mining rewards would incentivize people to create higher-quality flips and discourage farmers from using templates in flips, which could ultimately have a detrimental impact on the entire farming business.

**Reply by teavar** [reply]
@do2050 something has to be done about these type of users: 

https://scan.idena.io/identity/0x51425B61c0946feD811Fc1CCb4250625A16f2c7c#flips

I propose: 4 or more reported flips within 2 consecutive epochs = suspended

**Reply by illusiontorment** [reply]
At this point, it would be really justified to introduce compulsory explanation on why someone reports a flip, and of course a dispute resolution mechanism.

I really strive to create unique and high-quality flips every epoch, but I often get reported because I don't include the literal keyword (e.g. horns as "boney outgrows on the head") but a figurative form. It is bad enough to loose thousands of iDNA in a month due to the loss of validation rewards, please don't fuck with my mining rewards!

**Reply by do2050** [reply]
Did you know this epoch i presented very good flips and yet one of them took 4 reports and he was the best of them and all the words are there., Random reports

**Reply by teavar** [reply]
@do2050 Can you post a screenshot and a link to the flip?

**Reply by teavar** [reply]
@illusiontorment  imho a dispute resolution mechanism will require a lot of programming effort. 

Please post screenshots and links to the flips which you consider unfairly reported

**Comment by 0JustinMiles0** [comment]
As long as the report system provides a mediocre accuracy (reportable flips being not reported and non-reportable flips being reported), we should be careful about increasing the rewards of the reporters and penalty of the reported. Especially, if report rewards becomes quadratic, many identities won't have an incentive to report anymore. If the goal is to penalize flips created by flip farms I would suggest to consider this proposal: https://github.com/idena-network/idena-docs/discussions/134 which would prevent identities with a score below 92% to participate in the report session. This would remove a higher share of identities linked to flip farms as those struggle more to maintain a high share of identities with a score above 92%.

**Reply by teavar** [reply]
>  if report rewards becomes quadratic

Very bad idea indeed

**Reply by faket0shi** [reply]
Since I have heard this argument several times, I would like to know where they get the numbers to conclude that good flips are being reported.

**Comment by ltraveler** [comment]
@0JustinMiles0, quite an interesting point of view. However, it could delve even deeper, depending on identity status, the weight of the report vote could vary. Furthermore, we could incorporate the entire statistics of all created flips, using the proportion between created and bad (reported) flips to adjust the weight.

The reward mechanism for individuals on the report committee could be based on the validation fund of the reported identity, with the proportion determined by their stake (quadratic coefficient) and the points each identity accumulates to ensure the success of the report. In other words, a flip could be reported not based on consensus but rather on the threshold of points required for it to be considered as reported.

For example, we could establish a scale of 0 to 10 points for each report, taking into account factors such as the status and age of the identity, the ratio of good to bad flips, the frequency of flip reports by the identity, and the success rate of those reports. If the proportion leans towards false reports, the weight of the report could be relatively low.

For example, if we consider a report committee of each flip as 10 identities, and each identity can contribute a maximum of 10 points to the report total value, then the total maximum report points would be 100. This scale provides a robust framework for conducting a thorough assessment of flip quality. Furthermore, such a scale ensures that the rewards we intend to distribute will be allocated to deserving identities based on their individual contributions and personal rating levels.

Certainly, with such a finely-tuned scale, we could establish a specific threshold for reducing the mining rewards of identities, contingent upon the number of report points assigned to each identity's flip. Based on this threshold, we could calculate the percentage of mining rewards allocated to the report committees that participated in assessing each flip. This calculation would take into consideration the previously mentioned factors, if the reported identity continues to engage in mining during the epoch.

So, as a summary, the report reward fund would be funded by a validation rewards of the reported identity and a percentage of the mining rewards of the reported identity determined by the reporting scale based on the report composite score

**Comment by teavar** [comment]
reanimating this thread

- the results of the "Mining rights for reported Identities" oracle show that the majority agrees that a change is needed

https://discord.com/channels/634481767352369162/1255583086213730424/1264900056730046464

- imho this change is urgent

rough summary of the proposal:

3 created flips: 2+ reported flips = suspended
4 created flips: 2+ reported flips = suspended
5 created flips: 3+ reported flips = suspended

*A separate (or perhaps no new) rule should be applied to newbies since they can't be suspended
